cmsplugin_cascade/plugin_base.py

```diff
@@ -294,15 +294,24 @@ def save_model(self, request, new_obj, form, change):
                     new_obj.glossary[key] = old_obj.glossary[key]
         super(CascadePluginBase, self).save_model(request, new_obj, form, change)
 
-    def get_parent_instance(self, request=None):
+    def get_parent_instance(self, request=None, obj=None):
         """
         Get the parent model instance corresponding to this plugin. When adding a new plugin, the
         parent might not be available. Therefore as fallback, pass in the request object.
         """
-        try:
-            parent_id = self.parent.id
-        except AttributeError:
-            parent_id = request.GET.get('plugin_parent') if request else None
+        if obj is not None and hasattr(obj, 'parent_id'):
```

Comment:
previously I used try/except. This is the preferred way in Python. Please see EAFP in https://docs.python.org/3.5/glossary.html